src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -350,15 +362,12 @@ public int Match(object[] givenParameters)
                         if (_parameterValues[applyIndex] == null &&
                             _parameters[applyIndex].ParameterType.IsAssignableFrom(givenType))
                         {
+                            applyExactLength++;
```

Comment:
It might make sense to rename `applyExactLength` to a better name now. I don't think that matches what it is used for now.